### PR TITLE
Allow user to set auth headers per pull

### PR DESCRIPTION
**Pull request by davidxia**

### CODE REVIEW

File: src/main/java/com/spotify/docker/client/DefaultDockerClient.java [hunk 1]
```diff
@@ -657,6 +657,35 @@ public void pull(final String image, final ProgressHandler handler)
   }
 
   @Override
+  public void pull(String image, AuthConfig authConfig)
```
Comment (rohansingh) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: params should be `final`

File: src/main/java/com/spotify/docker/client/DefaultDockerClient.java [hunk 2]
```diff
@@ -657,6 +657,35 @@ public void pull(final String image, final ProgressHandler handler)
...
+      throws DockerException, InterruptedException {
+    pull(image, authConfig, new LoggingPullHandler(image));
+  }
+
+  @Override
+  public void pull(String image, AuthConfig authConfig, ProgressHandler handler)
```
Comment (rohansingh) [inline src/main/java/com/spotify/docker/client/DefaultDockerClient.java]: params here should be `final` too

**Comment by rohansingh** [comment]
:+1: *

*minor comments inline

**Comment by davidxia** [comment]
@rohansingh yay tests finally passed, try locally too for good measure?

**Comment by rohansingh** [comment]
All tests pass locally for me.

Works On My Machine award

**Pull request merged**